spring-integration-core/src/main/java/org/springframework/integration/gateway/MessagingGatewaySupport.java

```diff
@@ -333,6 +352,14 @@ private Object doSendAndReceive(Object object, boolean shouldConvert) {
 					error = ((ErrorMessage) reply).getPayload();
 				}
 			}
+			if (reply == null && this.errorOnTimeout) {
```

Comment:
@garyrussell Thank you very much for the swift and extremely helpful reply. It solved our problem :smile: